src/components/ui/RadioButtonGroup/RadioButtonGroup.tsx

```diff
@@ -0,0 +1,59 @@
+import { forwardRef, Fragment, useState } from "react";
+
+import { cn } from "@/utils/helper";
+import radioCheck from "@/assets/icons/radio-check.svg";
+import Icon from "@/components/ui/Icon";
+import { IRadioButtonGroupProps } from "@/components/ui/RadioButtonGroup/RadioButtonGroup.d";
+
+const RadioButtonGroup = forwardRef<HTMLInputElement, IRadioButtonGroupProps>(
+  (props, ref) => {
+    const { radioButtons, ...rest } = props;
+    const [selectedValue, setSelectedValue] = useState(rest.defaultValue);
+
+    return (
+      <Fragment>
+        {radioButtons.map((radioButton, index) => {
+          return (
+            <span
+              key={index}
```

Comment:
Adding straight `index` as `key` value must be avoided and suggested greater react community. This may hinder the efficiency of React's diffing algorithm. Better to bind the context (in word) with the index value to make unique throughout the application.

Reply:
Noted bhai